### ISSUE TITLE
Bug: playground is broken after opening link with the .js file type.

### ISSUE THREAD

**Issue by nayounsang** [typescript-eslint/typescript-eslint#12764]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.
- [x] If I used AI assistance to prepare this issue, I have reviewed and followed the [AI Contribution Policy](https://typescript-eslint.io/contributing/ai-policy/).

### Relevant Package

website

### Playground Link

https://typescript-eslint.io/play/#ts=5.7.2&fileType=.js&code=MYewdgzgLgBAhjAvDAjABgNwCgg&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
Check it out in the playground.
```

### ESLint Config

```javascript
{
  "rules": {}
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

- It must be possible to switch tabs and configure options

### Actual Result

https://github.com/user-attachments/assets/ddb2e7bb-66c3-48d8-a3c4-b50c25918134

- The tsconfig and eslintrc editors cannot be edited.
- The problem persists after changing the file type to .ts. a page refresh is required before the editor works.

### Additional Info

Following up to [review](https://github.com/typescript-eslint/typescript-eslint/pull/12744#discussion_r3849939204) in #12744

Opening a [Playground link](https://typescript-eslint.io/play/#ts=5.7.2&fileType=.ts&code=MYewdgzgLgBAhjAvDAjABgNwCgg&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false) in TS format and then switch the file type to .js successfully.
The issue does not appear to depend on other options, such as the TypeScript version or configuration.

#### Possible cause

If open a link with file type is `js`, `createParser` in website is called with initial `compilerOptions` and it is empty(`{}`).
And TypeScript throws an error indicating that allowJs must be enabled, causing the playground to fail to run.

### Versions

This issue affects the repository's website playground and is not dependent on the published package versions.

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `X.Y.Z` |
| `@typescript-eslint/parser`            | `X.Y.Z` |
| `@typescript-eslint/rule-tester`       | `X.Y.Z` |
| `@typescript-eslint/scope-manager`     | `X.Y.Z` |
| `@typescript-eslint/typescript-estree` | `X.Y.Z` |
| `@typescript-eslint/type-utils`        | `X.Y.Z` |
| `@typescript-eslint/utils`             | `X.Y.Z` |
| `TypeScript`                           | `X.Y.Z` |
| `ESLint`                               | `X.Y.Z` |
| `node`                                 | `X.Y.Z` |
